packages/api/src/modules/demandes_subventions/demandes_subventions.service.ts

```diff
@@ -1,38 +1,59 @@
-import { Siren, Siret } from "@api-subventions-asso/dto";
+import { AssociationIdentifiers, StructureIdentifiers } from '../../@types';
+import { isRna, isSiren, isSiret } from '../../shared/Validators';
 import providers from "../providers";
-import DemandesSubventionsProvider from "./interfaces/DemandesSubventionsProvider";
-import DemandeSubvention from "./interfaces/DemandeSubvention";
+import DemandesSubventionsProvider from "./@types/DemandesSubventionsProvider";
+import DemandeSubvention from "./@types/DemandeSubvention";
+import { StructureIdentifiersEnum } from "../../@enums/StructureIdentifiersEnum"
+import { siretToSiren } from '../../shared/helpers/SirenHelper';
+import { capitalizeFirstLetter } from '../../shared/helpers/StringHelper'
+import { Siret } from '@api-subventions-asso/dto';
 
 export class DemandesSubventionsService {
 
-    async getDemandeSubventionsBySiret(siret: Siret) {
-        const data = await (await this.aggregate(siret, "SIRET")).filter(asso => asso) as DemandeSubvention[];
-
-        if (!data.length) return null;
-
+    async getByAssociation(id: AssociationIdentifiers) {
+        const type = this.getIdentifierType(id) ;
+        if (!type) throw new Error("You must provide a valid SIREN or RNA");
+        const data = await (await this.aggregate(id, type))?.filter(asso => asso) as DemandeSubvention[];
+        if (!data.length) throw  new Error("Association not found");
```

Comment:
Un espace en trop entre throw et new Error

Reply:
Je me demande si on ne devrait pas avoirs des erreurs type genre AssoError ou un truc du genre, ça éviterai d'avoir plusieurs fois des throw new Error ("asso not found") dans le futur.

Reply:
Oui c'est une idée d'avoir des Erreurs custom, à voir l'utilité. Si c'est utilisé qu'une seule fois, pas nécessaire